satpy/tests/reader_tests/test_agri_l1.py

```diff
@@ -255,6 +257,14 @@ def teardown(self):
         """Stop wrapping the HDF5 file handler."""
         self.p.stop()
 
+    def test_times_correct(self):
+        """Test that the reader handles the two possible time formats correctly."""
+        reader = self._create_reader_for_resolutions(1000)
+        timestamp_ms = reader.start_time.timestamp()
+        reader = self._create_reader_for_resolutions(2000)
+        timestamp_noms = reader.start_time.timestamp()
+        np.testing.assert_almost_equal(timestamp_ms - timestamp_noms, 0.807)
+
```

Comment:
This test seems a little confusing and indirect. Could you make this two separate tests and check that the returned time is as expected. Or even use `pytest.mark.parametrize` and have the input resolution and the expected time as parameters to the test function. However, I see that this class uses `setup` and `teardown` methods but the class isn't based on unittest, I think these need to be changed to `setup_method` and `teardown_method` (they only work now due to a backwards compatibility of pytest with old nose tests).

Reply:
Done!